QUICK-FIX Handle incorrect data payload on object creation

mashabarashko opened the pull request:
<!-- If your PR depends on other tickets or PRs, you can list them here
# Dependencies

This PR is `on hold` until the dependencies are merged:

- [ ] GGRC-1234
- [ ] #1234
-->

# Issue description

This is additional PR for issue GGRC-6538. Request payload on task group post was incorrect (id should be a number instead of object). Response status code should be equal 400 in such case.

# Sanity checklist

- [ ] I have clicked through the app to make sure my changes work and not break the app.
- [x] I have applied the correct milestone and labels.
- [x] My changes fix the issue described in the description (and do nothing else). 🤞
- [x] My changes are covered by tests.
- [x] My changes follow our [performance guidelines](https://github.com/google/ggrc-core/blob/dev/docs/source/contributing/performance.rst).
- [x] My changes follow our [js](https://github.com/google/ggrc-core/blob/dev/docs/source/contributing/javascript.rst) and/or [python](https://github.com/google/ggrc-core/blob/dev/docs/source/contributing/python.rst) guidelines.
- [x] My commits follow our [commit guidelines](https://github.com/google/ggrc-core/blob/dev/docs/source/contributing/git/how_to_write_a_commit_message.rst).

<!-- If your PR includes a migration include the additional checklist items
# Migration checklist
- [ ] Migration passes all checks from our [PR review guidelines](https://github.com/google/ggrc-core/blob/dev/docs/source/contributing/git/reviewing_pull_requests.rst#reviewing-a-pr-containing-database-migration-scripts)
- [ ] Upon merging, add 'check migration chain' and 'kokoro:run' labels to all open PRs with a label 'migration'
-->

# PR Review checklist

- [ ] The changes fix the issue and don't cause any apparent regressions.
- [ ] Labels and milestone are correctly set.
- [ ] The solution description matches the changes in the code.
- [ ] There is no apparent way to improve the performance & design of the new code.
- [ ] The pull request is opened against the correct base branch.
- [ ] Upon merging, the Jira ticket's fixversion is correctly set and the ticket is moved to "QA - In Progress".

<!-- If your code is not finished yet can include a TODO check list
# TODO

- [ ] First item on the TODO
-->

mashabarashko commented:
`needs work` because FE fix should be provided.

mashabarashko commented:
@NikitaKulik It's used only in common._build_request_stub_cache, which is used in collection_post, common.collection_post and common.put. All of them are depend on UI request (please, correct me if I'm wrong and provide an example). I think this method is a better place to check payload provided this is recursive function, and format of payload can be difficult enough.

belovmd commented on src/ggrc/services/common.py:
```diff
@@ -999,7 +999,11 @@ def _gather_referenced_objects(self, data, accomulator=None):
       if "type" in data and data.get("id"):
-        accomulator[data["type"]].add(data["id"])
+        try:
+          accomulator[data["type"]].add(data["id"])
```
please fix misprint in `accumulator` word

belovmd commented on test/integration/ggrc_workflows/test_workflow_api_calls.py:
```diff
@@ -400,6 +400,31 @@ def test_create_task_group(self):
     self.assertEqual(response.status_code, 201)
 
+  def test_incorrect_id_in_payload(self):
+    """Test that 400 is raised for payload
```
Please correct this docstring according to our codestyle.

belovmd commented on test/integration/ggrc_workflows/test_workflow_api_calls.py:
```diff
@@ -400,6 +400,31 @@ def test_create_task_group(self):
     response = self.api.post(all_models.TaskGroup, data)
     self.assertEqual(response.status_code, 201)
 
+  def test_incorrect_id_in_payload(self):
```
is it possible to create test for incorrect type?

mashabarashko commented:
@belovmd I've fixed code style issues, but I wasn't able to fix this new test, I wasn't able to reproduce the situation with incorrect type value in tests and removed it. Please, review.
